PWGHF/TableProducer/candidateCreator2Prong.cxx

```diff
@@ -150,24 +175,11 @@ struct HfCandidateCreator2Prong {
     // loop over pairs of track indices
     for (const auto& rowTrackIndexProng2 : rowsTrackIndexProng2) {
 
-      // reject candidates in collisions outside the centrality range
+      /// reject candidates not satisfying the event selections
       auto collision = rowTrackIndexProng2.template collision_as<Coll>();
-      float centrality = -1.;
-      if constexpr (centEstimator != CentralityEstimator::None) {
-        if constexpr (centEstimator == CentralityEstimator::FT0C) {
-          centrality = collision.centFT0C();
-        } else if constexpr (centEstimator == CentralityEstimator::FT0M) {
-          centrality = collision.centFT0M();
-        } else {
-          LOGP(fatal, "Centrality estimator different from FT0C and FT0M, fix it!");
-        }
-        if (centrality < centralityMin || centrality > centralityMax) {
-          continue;
-        }
-      }
-
-      /// event selection: sel8, PV posZ, TF border cut
-      if (!isHfCollisionSelected(collision, useSel8Trigger, maxPvPosZ, useTimeFrameBorderCut)) {
+      const uint16_t statusCollision = isHfCollisionSelected<centEstimator>(collision, std::array<float, 2>{centralityMin.value, centralityMax.value}, useSel8Trigger, maxPvPosZ, useTimeFrameBorderCut);
+      if (statusCollision != 0) {
+        /// at least one event selection not satisfied --> reject the candidate
         continue;
       }
```

Comment:
```suggestion
      const auto rejectionMask = getHfCollisionRejectionMask<centEstimator>(collision, std::array<float, 2>{centralityMin.value, centralityMax.value}, useSel8Trigger, maxPvPosZ, useTimeFrameBorderCut);
      if (statusCollision != 0) {
        /// at least one event selection not satisfied --> reject the candidate
        continue;
      }
```

Reply:
done

Reply:
You are not using `auto`. If the return value `getHfCollisionRejectionMask` changes, you will have to change the variable type everywhere.